give initial states more attention

User Doc is only using docker envs, his startup log looks fine, but the line:
`[ERROR] 2022/07/16 20:33:59 Getting initial service states from Notifiarr.com: converting response values to json: json: cannot unmarshal array into Go struct field .fields of type map[string][]uint8`
He doesn't has any services enabled, so there is no initial state, not sure if that is a Nit or a Cap thing to fix.

Origin user: Gabe
Origin message: [Discord Server Link](https://discord.com/channels/764440599066574859/774710801332305971/997966365824663702)
Reporting user: TwitchCaptain
Reporting message: [Discord Server Link](https://discord.com/channels/764440599066574859/774710801332305971/998029812184662056)

```
[INFO] 2022/07/16 20:33:58 notifiarr v0.3.1-530 Starting! [PID: 1] 2022-07-16 20:33:58.253784982 +0000 UTC m=+0.121897894
[INFO] 2022/07/16 20:33:58 ==> Using env variables only. Config file not found.
[INFO] 2022/07/16 20:33:59 ==> Unique ID: xxx (notifiarr)
[INFO] 2022/07/16 20:33:59 ==> GoLift Discord: https://golift.io/discord <==
[INFO] 2022/07/16 20:33:59 ==> Production Startup Settings <==
[INFO] 2022/07/16 20:33:59 => Lidarr Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Prowlarr Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Radarr Config: 1 server: http://radarr:7878/ apikey:true timeout:1m verify_ssl:false stuck_items:false corrupt:false backup:false
[INFO] 2022/07/16 20:33:59 => Readarr Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Sonarr Config: 1 server: http://sonarr:8989/ apikey:true timeout:1m verify_ssl:false stuck_items:false corrupt:false backup:false
[INFO] 2022/07/16 20:33:59 => Deluge Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Qbit Config: 0 servers
[INFO] 2022/07/16 20:33:59 => SABnzbd Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Plex Config: 1 server: Nas @ https://plex.name/ (enables incoming APIs and webhook)
[INFO] 2022/07/16 20:33:59 => Tautulli Config (enables name map): 0 servers
[INFO] 2022/07/16 20:33:59 => MySQL Config: 0 servers
[INFO] 2022/07/16 20:33:59 => Timeout: 1m, Quiet: false
[INFO] 2022/07/16 20:33:59 => Trusted Upstream Networks: (none)
[INFO] 2022/07/16 20:33:59 => Web HTTP Listen: http://0.0.0.0:5454/
[INFO] 2022/07/16 20:33:59 ==> System Snapshot Collection Started, interval: 1h, timeout: 30s, enabled: cpu, load, memory, uptime, users, temps
[INFO] 2022/07/16 20:33:59 ==> Plex Sessions Collection Started, URL: https://plex.name/, interval:30m timeout:10s webhook_cooldown:10s delay:10s
[INFO] 2022/07/16 20:33:59 ==> Keeping 86 Radarr Custom Formats and 0 Sonarr Release Profiles synced, interval:15m
[INFO] 2022/07/16 20:33:59 ==> Sending Current State Data for Dashboard, interval:30m, serial:false
[INFO] 2022/07/16 20:33:59 ==> Custom Timers Enabled: 1 timers provided
[INFO] 2022/07/16 20:33:59 ==> Started 16 Notifiarr Timers with 18 Triggers
[ERROR] 2022/07/16 20:33:59 Getting initial service states from Notifiarr.com: converting response values to json: json: cannot unmarshal array into Go struct field .fields of type map[string][]uint8
[INFO] 2022/07/16 20:33:59 ==> Service Checker Started! 0 services, interval: 10m, parallel: 1
[INFO] 2022/07/16 20:33:59 Sent 0 service check states to Notifiarr, event: start, buffer: 0/1000. Website took 0s and replied with: success, "Client uptime updated but network integration stopped, no services sent"
```

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -1,3 +1,12 @@
+issues:
+  exclude-rules:
+    # Exclude funlen for testing files.
+    - linters:
+        - nonamedreturns
+        - wsl
+        - cyclop
+        - nolintlint
+      path: 'cmd/'
 linters:
   enable-all: true
   disable:

--- a/go.mod
+++ b/go.mod
@@ -6,7 +6,7 @@ go 1.17
 require (
 	golift.io/cnfg v0.2.1
 	golift.io/cnfgfile v0.0.0-20220509075834-08755d9ef3f5
-	golift.io/deluge v0.10.0
+	golift.io/deluge v0.10.1-0.20220720064250-90bc206753c9
 	golift.io/nzbget v0.1.4-0.20220719081244-8d4d909649d3
 	golift.io/qbit v0.0.0-20220709224202-8d3d80f96de0
 	golift.io/rotatorr v0.0.0-20220126065426-d6b5acaac41c

--- a/pkg/bindata/templates/index.html
+++ b/pkg/bindata/templates/index.html
@@ -8,6 +8,7 @@
                 <div class="brand clearfix" style="z-index:100;">
                     <a class="nav-link" href="#landing" onclick="swapNavigationTemplate('landing')"><!-- outside the h4 to keep it white -->
                         <h4 class="text-uppercase text-white" style="margin:10px 0px 0px 10px">
+                            <span class="pending-change-container" style="display:none;"><i class="fas text-danger fa-exclamation-circle" title="Unsaved Changes!" style="display:none;"></i></span>
                             <img src="{{files}}/images/logo/notifiarr.png" style="height:47px;">
                             <span class="mobile-hide">Notifiarr Client</span>
                         </h4>
@@ -19,7 +20,9 @@
                     <a class="nav-button text-danger fas fa-power-off" title="Shut down application (quit)." onClick="shutDownApp();" style="margin-right:0;"></a>
                     <a class="nav-button text-success fas fa-recycle" title="Reload application configuraton and refresh." onClick="reloadConfig();"></a>
                     {{- end}}
-                    <span class="menu-btn pull-left" style="font-size:1.3em;"><i class="fa fa-bars"> <span class="pending-change-container" style="display:none;"><i class="fas text-danger fa-exclamation-circle" title="Unsaved Changes!" style="display:none;"></i></span></i></span>
+                    <span class="menu-btn pull-left" style="font-size:1.3em;"><i class="fa fa-bars"><span class="pending-change-container" style="display:none;">
+                    </span></i>
+                    </span>
                     <ul class="ts-profile-nav"><div style="margin-top:25px; height:40px;">&nbsp;</div></ul>
                 </div>
 

--- a/pkg/bindata/templates/index.html
+++ b/pkg/bindata/templates/index.html
@@ -6,9 +6,13 @@
         <div class="page-container">
             <div class="page-body">
                 <div class="brand clearfix" style="z-index:100;">
+                    <span class="mobile-hide tablet-hide" title="Unsaved Changes!">
+                        <span class="pending-change-container" style="display:none;margin:10px 0px 0px 10px;font-size: 18px;">
+                            <i class="fas text-danger fa-exclamation-circle" style="display:none;"></i>
+                        </span>
+                    </span>
                     <a class="nav-link" href="#landing" onclick="swapNavigationTemplate('landing')"><!-- outside the h4 to keep it white -->
                         <h4 class="text-uppercase text-white" style="margin:10px 0px 0px 10px">
-                            <span class="pending-change-container" style="display:none;"><i class="fas text-danger fa-exclamation-circle" title="Unsaved Changes!" style="display:none;"></i></span>
                             <img src="{{files}}/images/logo/notifiarr.png" style="height:47px;">
                             <span class="mobile-hide">Notifiarr Client</span>
                         </h4>
@@ -20,14 +24,18 @@
                     <a class="nav-button text-danger fas fa-power-off" title="Shut down application (quit)." onClick="shutDownApp();" style="margin-right:0;"></a>
                     <a class="nav-button text-success fas fa-recycle" title="Reload application configuraton and refresh." onClick="reloadConfig();"></a>
                     {{- end}}
-                    <span class="menu-btn pull-left" style="font-size:1.3em;"><i class="fa fa-bars"><span class="pending-change-container" style="display:none;">
-                    </span></i>
+                    <span class="menu-btn pull-left" style="font-size:1.3em;">
+                        <i class="fa fa-bars">
+                            <span class="pending-change-container" style="display:none;">
+                                <i class="fas text-danger fa-exclamation-circle" title="Unsaved Changes!" style="display:none;"></i>
+                            </span>
+                        </i>
                     </span>
                     <ul class="ts-profile-nav"><div style="margin-top:25px; height:40px;">&nbsp;</div></ul>
                 </div>
 
                 <div class="ts-main-content">
-                    <nav class="ts-sidebar">
+                    <nav class="ts-sidebar" style="margin-bottom: 25px;">
                         <ul class="ts-sidebar-menu">
                         {{- if .Username }}
                             {{- if .Dynamic }}

--- a/pkg/client/instance_tests.go
+++ b/pkg/client/instance_tests.go
@@ -124,13 +124,12 @@ func (c *Client) testInstance(response http.ResponseWriter, request *http.Reques
 }
 
 func testDeluge(ctx context.Context, config *deluge.Config) (string, int) {
-	if deluge, err := deluge.NewNoAuth(config); err != nil {
+	deluge, err := deluge.New(ctx, config)
+	if err != nil {
 		return "Connecting: " + err.Error(), http.StatusBadGateway
-	} else if xfers, err := deluge.GetXfersContext(ctx); err != nil {
-		return "Getting Transfers: " + err.Error(), http.StatusBadGateway
-	} else {
-		return fmt.Sprintf("Connection Successful! %d Transfers", len(xfers)), http.StatusOK
 	}
+
+	return fmt.Sprintf("Connection Successful! Version: %s", deluge.Version), http.StatusOK
 }
 
 func testNZBGet(ctx context.Context, config *nzbget.Config) (string, int) {

--- a/pkg/services/services.go
+++ b/pkg/services/services.go
@@ -207,10 +207,12 @@ func (c *Config) Running() bool {
 	return c.stopChan != nil
 }
 
-// Stop ends all service checker routines.
+// Stop sends current states to the website and ends all service checker routines.
 func (c *Config) Stop() {
 	c.stopLock.Lock()
 	defer c.stopLock.Unlock()
+
+	c.updateStatesOnSite()
 
 	if c.stopChan == nil {
 		return

--- a/pkg/snapshot/diskusage.go
+++ b/pkg/snapshot/diskusage.go
@@ -40,6 +40,10 @@ func (s *Snapshot) getDisksUsage(ctx context.Context, run bool, allDrives bool) 
 			continue
 		}
 
+		if usage.Used == 0 && usage.Free > 0 && usage.Total > usage.Free {
+			usage.Used = usage.Total - usage.Free
+		}
+
 		s.DiskUsage[partitions[idx].Device] = &Partition{
 			Device: partitions[idx].Mountpoint,
 			Total:  usage.Total,

--- a/pkg/snapshot/diskusage.go
+++ b/pkg/snapshot/diskusage.go
@@ -11,7 +11,7 @@ import (
 	"github.com/shirou/gopsutil/v3/disk"
 )
 
-func (s *Snapshot) getDisksUsage(ctx context.Context, run bool, allDrives bool) []error {
+func (s *Snapshot) getDisksUsage(ctx context.Context, run bool, allDrives bool) []error { // nolint:cyclop
 	if !run {
 		return nil
 	}

